lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -1321,14 +1321,15 @@ void LCIStrategicPlugin::plan_maneuvers_callback(
   }
 
   // get the lanelet that is on the route in case overlapping ones found
-  for (auto llt : current_lanelets)
-  {
-    auto route = wm_->getRoute()->shortestPath();
-    if (std::find(route.begin(), route.end(), llt) != route.end())
-    {
-      current_lanelet = llt;
-      break;
-    }
+  auto llt_on_route_optional = wm_->getFirstLaneletOnShortestPath(current_lanelets);
+
+  if (llt_on_route_optional){
+    current_lanelet = llt_on_route_optional.value();
+  }
+  else{
+    RCLCPP_WARN_STREAM(rclcpp::get_logger("plan_delegator"), "When adjusting maneuver for maneuvers other than lane follow, no previous lanelet found on the shortest path. " <<
```

Comment:
Need to change to `rclcpp::get_logger("lci_strategic_plugin")` and also adjust the log text for this logic, since it is different than plan_delegator's. It looks like we are assigning the vehicle's current lanelet based on the plan maneuver request's `req->veh_x` and `req->veh_y` coordinates.

Reply:
Fixed the package name and description